src/main/java/address/MainApp.java

```diff
@@ -73,45 +74,29 @@ protected void initComponents(Config config, UserPrefs userPrefs) {
         syncManager = new SyncManager(config);
         keyBindingsManager = new KeyBindingsManager();
         updateManager = new UpdateManager();
-        alertMissingDependencies();
-        //TODO: should this be here? looks out of place
     }
 
     @Override
     public void start(Stage primaryStage) {
-        logger.info("Starting application: {}", Version.getCurrentVersion());
+        logger.info("Starting application: {}", MainApp.VERSION);
+        DependencyChecker dependencyChecker = new DependencyChecker(this::quit);
+        dependencyChecker.run();
```

Comment:
If it is in init method, I will need to use an alert dialog that is not from JavaFX but from Java Swing instead because at that point in that the UI-thread has not been created. Is that alright?